public/app/plugins/datasource/tempo/resultTransformer.ts

```diff
@@ -85,8 +91,10 @@ export function transformTraceList(
   datasourceName: string,
   traceRegexs: string[]
 ): DataQueryResponse {
-  const frame = createTableFrame(response.data[0], datasourceId, datasourceName, traceRegexs);
-  response.data[0] = frame;
+  response.data.forEach((data, index) => {
+    const frame = createTableFrame(data, datasourceId, datasourceName, traceRegexs);
```

Comment:
Which change btw prevents showing log panel? Seems like we were already returning table frame here.

Reply:
yes but only for the first frame and all the other had `preferredVisualisationType: 'logs',`